lib/matplotlib/axes/_base.py

```diff
@@ -3837,9 +3839,11 @@ def set_ylabel(self, ylabel, fontdict=None, labelpad=None, *,
                    else mpl.rcParams['yaxis.labellocation'])
         _api.check_in_list(('bottom', 'center', 'top'), loc=loc)
         if loc == 'bottom':
-            kwargs.update(y=0, horizontalalignment='left')
+            kwargs.update(y=0, verticalalignment='bottom')
+        elif loc == 'center':
+            kwargs.update(y=0.5, verticalalignment='center')
         elif loc == 'top':
-            kwargs.update(y=1, horizontalalignment='right')
+            kwargs.update(y=1, verticalalignment='top')
```

Comment:
I don't think this change is correct, is it?  horizontal alignment is relative to spinning 90 degrees I think.